### PR TITLE
[VL] Support change columnar shuffle commpression level

### PR DESCRIPTION
**Pull request by ulysses-you**
## What changes were proposed in this pull request?

Vanilla Spark support change shuffle commpression level if the codec is zstd. This pr makes columnar shuffle follow the behavior. It respects config `spark.io.compression.zstd.level`.

## How was this patch tested?

manually test a small dataset

level=1
image

level=21
image

### CODE REVIEW

File: cpp/core/shuffle/Options.h [hunk 1]
```diff
@@ -56,6 +57,7 @@ struct PartitionWriterOptions {
   int32_t compressionThreshold = kDefaultCompressionThreshold;
   arrow::Compression::type compressionType = arrow::Compression::LZ4_FRAME;
   CodecBackend codecBackend = CodecBackend::NONE;
+  int32_t compressionLevel = std::numeric_limits<int>::min();
```
Comment (marin-ma) [inline cpp/core/shuffle/Options.h:60]: Is it possible to use `kUseDefaultCompressionLevel` as the default value here?

File: cpp/velox/shuffle/VeloxShuffleReader.cc [hunk 2]
```diff
@@ -268,162 +268,6 @@ std::shared_ptr<VeloxColumnarBatch> makeColumnarBatch(
   return std::make_shared<VeloxColumnarBatch>(std::move(rowVector));
 }
 
-std::shared_ptr<arrow::Buffer> readColumnBuffer(const arrow::RecordBatch& batch, int32_t fieldIdx) {
```
Comment (marin-ma) [inline cpp/velox/shuffle/VeloxShuffleReader.cc:271]: Thanks for the cleanup!

File: cpp/core/shuffle/Options.h (hunk 1 above)
Comment (ulysses-you) [reply cpp/core/shuffle/Options.h]: changed to use kUseDefaultCompressionLevel

File: cpp/core/shuffle/Options.h [hunk 3]
```diff
@@ -37,6 +37,7 @@ enum PartitionWriterType { kLocal, kCeleborn };
 struct ShuffleReaderOptions {
   arrow::Compression::type compressionType = arrow::Compression::type::LZ4_FRAME;
   CodecBackend codecBackend = CodecBackend::NONE;
+  int32_t compressionLevel = std::numeric_limits<int>::min();
```
[the remark this answers is not in the document]
Comment (ulysses-you) [reply cpp/core/shuffle/Options.h]: ah, I looked at the arrow zstd codec code, it's truth that the compression level would be ignored. Removed this.

**Review by marin-ma — APPROVED** [review]

**Comment by marin-ma** [comment]
Thanks!

**Pull request merged**